USE filtered VCF file from snpcalling

## IQTREE

Convert VCF to Phylip
```
vcf2phylip.py --input snpfiltered.recode.vcf
```

## add reference sequence to phylip 


first cut the positions of each snp from the vcf and put them in a bed file
```
awk 'BEGIN {OFS="\t"} !/^#/ {print $1, $2-1, $2}' all_samples.recode.vcf > allsample_vcf_positions.bed
```
need to convert from 1base to 0base so do $2-1 - remove 1 from the position

now to frankenstein together some code from gemma/internet:

```
	## extract BED positions from masked genome fasta
bedtools getfasta -fi gb_hic_purged.fa.masked -bed allsample_vcf_positions.bed -fo reference_extracted_sequences.fasta

	## the resulting file has a header and each base on a different line
	##remove headers - resulting file still has a base on each line
awk '/^>/ {if (seq) print seq; seq=""; next} {seq=seq $0} END {print seq}' reference_extracted_sequences.fasta > refseq_string.txt

	##stick all the lines together - creates a little loop in sed to loop back to 'a' until the last line of the file ($!)
	##and tells it to put the next line up with the first and remove all newline characters then sticks Tasmanian_reference_GB in front 
sed ':a;N;$!ba;s/\n//g' refseq_string.txt | sed "s/^/Tasmanian_reference_GB\t/g" > refseq_string_row.txt
```
now we have a file with a name and then the reference sequence genotypes on it that we can cat on to the end of our phylip file
```
python vcf2phylip.py --input all_samples.recode.vcf

cat refseq_string_row.txt >> all_samples.recode.min4.phy
```
then manually change sample ## at top from 174 to 175


## Run IQtree
```
#!/bin/bash -e
#SBATCH --job-name=gbrepeatmodeler # Job name (shows up in the queue)
#SBATCH --time=24:00:00      # Walltime (HH:MM:SS) 48h
#SBATCH --mem=32G            # Memory in MB
#SBATCH --account=uoo02831
#SBATCH --cpus-per-task=16

module purge

IQ-TREE/2.2.2.2-gimpi-2022a

iqtree2 -nt 16 -s snpfiltered.recode.min4.phy -st DNA -m GTR+G -bb 1000  -pre galtestinferred

echo "tree complete"
```
-nt = cpus
-s = input data
-m = substitution model
-bb = # bootstraps
-pre = output prefix

efficiency report:
```
Job ID: 54307808
Cluster: mahuika
User/Group: iwias275/iwias275
State: COMPLETED (exit code 0)
Nodes: 1
Cores per node: 16
CPU Utilized: 5-08:15:35
CPU Efficiency: 93.80% of 5-16:44:00 core-walltime
Job Wall-clock time: 08:32:45
Memory Utilized: 3.20 GB
Memory Efficiency: 10.01% of 32.00 GB
```
can reduce memory and time next run

removing the significant snps to remake tree like jon suggested
```
vcftools --vcf maskedrefiltered.recode.vcf --out nosigfiltered --exclude-positions SIGNIFICANT_SNPS_FOR_TREERM.txt --max-missing 0.8 --recode
```
```
python3 vcf2phylip.py --input nosigfiltered.recode.vcf
```


## PCA -PLINK
Ludo gave me a way to do this but I also did it this way because R gave me a warning the way I did it was depreciated

idrk if this is right but it looks ok

```
module load PLINK
```
Convert VCF file with PLINK
```
plink2 --vcf snpfiltered.recode.vcf --make-bed --out test_plink_output
```

PCA analysis
```
plink2 --bfile test_plink_output --pca --out pca_results
```
Outputs eigenvector and eigenvalue files - use these to plot in R

### redo with refiltered snps

```
plink2 --vcf maskedrefiltered.recode.vcf --make-bed --out plink_maskedrefiltered_out
```
```
plink2 --bfile plink_maskedrefiltered_out --pca --out pca_refiltered_out
```
**insert rmd here**

refilter vcf to remove lagoon saddle tarn and rerun plink
```
nano pca_filtering_indv.txt

KOARO_Lag_K01
KOARO_Lag_K20
KOARO_Lag_k02
KOARO_Lag_K03
KOARO_Lag_K04
KOARO_Lag_K05
KOARO_Lag_K07
KOARO_Lag_K08
KOARO_Lag_K09
KOARO_Lag_K11
KOARO_Lag_K13
KOARO_Lag_K14
KOARO_Lag_K15
KOARO_Lag_K19
```

```
vcftools --vcf maskedrefiltered.recode.vcf --out pca_subset_refiltered --max-missing 0.8 --remove pca_filtering_indv.txt --recode
```

```
plink2 --vcf pca_subset_refiltered.recode.vcf --make-bed --out plink_outputs/subset_pca_lagrm_plinkout
```
```
mkdir subset_lagrm_pca

plink2 --bfile subset_pca_lagrm_plinkout --pca --out subset_lagrm_pca/subset_lagrm_pca_out
```

## Linkage analysis _LDBlockShow

download git rep and install?
```
  git clone https://github.com/hewm2008/LDBlockShow.git
        cd LDBlockShow ; chmod 755 configure  ;  ./configure;
        make;
        mv LDBlockShow  bin/;    #     [rm *.o]
```

plot:
```
./bin/LDBlockShow -InVCF  maskedrefiltered.recode.vcf -InGWAS GWAS_OUTPUT_FILE_FOR_LD.txt  -OutPut scaff17_ldblock_default_out  -Region  ptg000017l_1:222179:8840409 -NoShowLDist 90000000
```

with sig snps
```
./bin/LDBlockShow -InVCF  maskedrefiltered.recode.vcf \
  -InGWAS GWAS_OUTPUT_FILE_FOR_LD.txt  \
  -OutPut scaff17_ldblock_default_out  \
  -Region  ptg000017l_1:222179:8840409 \
  -NoShowLDist 90000000 \
  -SpeSNPName sig_snp_names.txt
```

resize a couple of points
```
./bin/ShowLDSVG --InPreFix test_scaff17_ldblock_default_out \
-InGWAS GWAS_OUTPUT_FILE_FOR_LD.txt \
-OutPut resized_scaff17_gwas_out \
-NoShowLDist 90000000 \
-SpeSNPName sig_snp_names.txt \
-SNPNameSizeRatio 0.5 \
-PointSizeRatio 0.5
```

without the snp points
```
./bin/ShowLDSVG --InPreFix scaff17_ldblock_default_out -InGWAS GWAS_OUTPUT_FILE_FOR_LD.txt \
  -OutPut nonames_resized_scaff17_gwas_out -NoShowLDist 90000000 -PointSizeRatio 0.5
```


### new rerun 
```
./bin/LDBlockShow -InVCF  ../subset_vcfs/diadromous_only_vcf_renamed.vcf \
  -InGWAS ../subset_vcfs/q_value_gwas.txt  \
  -OutPut ../ldblock_outputs/scaff21_linkage_withsig  \
  -Region  SCAFFOLD_0021:222179:8840409 \
  -NoShowLDist 90000000 \
  -SpeSNPName ../subset_vcfs/significant_snps_NUMBERED.txt
```

resize a couple of points
```
./bin/ShowLDSVG --InPreFix ../ldblock_outputs/scaff21_linkage_withsig \
-InGWAS ../subset_vcfs/fst_gwas.txt \
-OutPut resized_scaff21_withsig \
-NoShowLDist 90000000 \
-SpeSNPName ../subset_vcfs/significant_snps_NUMBERED.txt \
-SNPNameSizeRatio 0.5 \
-PointSizeRatio 0.5
```

without the snp points
```
./bin/ShowLDSVG --InPreFix scaff17_ldblock_default_out -InGWAS GWAS_OUTPUT_FILE_FOR_LD.txt \
  -OutPut nonames_resized_scaff17_gwas_out -NoShowLDist 90000000 -PointSizeRatio 0.5
```

```
./bin/LDBlockShow -InVCF  ../subset_vcfs/noislands_renamed_full_sorted.vcf \
  -InGWAS ../subset_vcfs/q_value_gwas.txt  \
  -OutPut ../ldblock_outputs/fulldata_noislands_linkage  \
  -Region  SCAFFOLD_0021:222179:8840409 \
  -NoShowLDist 90000000
 -SpeSNPName ../subset_vcfs/significant_snps_NUMBERED.txt
```
```
./bin/ShowLDSVG --InPreFix ../ldblock_outputs/fulldata_noislands_linkage -OutPut ../ldblock_outputs/fulldata_noislands_linkage_resized -NoShowLDist 90000000 -SpeSNPName ../subset_vcfs/significant_snps_NUMBERED.txt -SNPNameSizeRatio 0 -PointSizeRatio 0.5

```
FINAL
```
./bin/LDBlockShow -InVCF  ../subset_vcfs/noislands_renamed_full_sorted.vcf -OutPut ../ldblock_outputs/out  -Region scaffold_0021:222179:8840409  -NoShowLDist 90000000

./bin/ShowLDSVG -InPreFix ../ldblock_outputs/out -OutPut ../ldblock_outputs/out -SpeSNPName ../subset_vcfs/significant_snps_noname.txt -NoShowLDist 90000000 -PointSizeRatio 0.5

how to remove the snp names/...............
