```
module load BWA SAMtools picard
```

Parameters file (executed in ~/nobackup/source_files/hic/mapping/)
```
#!/bin/bash

SRA='TRIMMED_HiC_R'
LABEL='gb_hic_map'
IN_DIR='./hicdata'
REF='./hicdata/reordered_gb_hic_purged_masked'
FAIDX='$REF.fa.fai'
PREFIX='./hicdata/reordered_gb_hic_purged_masked'  ##prefix for index files
RAW_DIR='./raw_outdir'
FILT_DIR='./filt_outdir'
FILTER='./scripts/filter_five_end.pl'
COMBINER='./scripts/two_read_bam_combiner.pl'
STATS='./scripts/get_stats.pl'
PICARD='/opt/nesi/CS400_centos7_bdw/picard/2.26.10-Java-11.0.4/picard.jar'
TMP_DIR='./temp'
PAIR_DIR='./paired_outdir'
REP_DIR='./deduped_outdir'
REP_LABEL=${LABEL}_rep1
MAPQ_FILTER=10
CPU=4
```
Full paths were giving me grief coz im stupid so relatives it is

Wehn running scripts up cpu to relevant ## (usually 12)

```
chmod +x parameters.sh

. parameters.sh
```

Index genome (again)
```
samtools faidx reordered_gb_hic_purged_masked.fa
```

Create BWA index file:
```
$BWA index -a bwtsw -p $PREFIX $REF
```

Re-trim UMIs (just in case)
```
zcat Bruce_fish_HiC_S1_R1_001.fastq.gz | awk '{ if(NR%2==0) {print substr($1,6)} else {print} }' | gzip > TRIMMED_HiC_R1.fastq.gz

zcat Bruce_fish_HiC_S1_R2_001.fastq.gz | awk '{ if(NR%2==0) {print substr($1,6)} else {print} }' | gzip > TRIMMED_HiC_R2.fastq.gz
```

Align Hi-C reads to genome
 ```
#!/bin/bash -e
#SBATCH --job-name=bwa_mem
#SBATCH --time=15:00:00
#SBATCH --mem=16G
#SBATCH --account=uoo02831
#SBATCH --cpus-per-task=12

module load BWA SAMtools 

SRA='TRIMMED_HiC_R'
LABEL='gb_hic_map'
IN_DIR='./hicdata'
BWA='/opt/nesi/CS400_centos7_bdw/BWA/0.7.18-GCC-12.3.0/bin/bwa'
SAMTOOLS='/opt/nesi/zen3/SAMtools/1.22-GCC-12.3.0/bin/samtools'
REF='./hicdata/reordered_gb_hic_purged_masked'
FAIDX='$REF.fai'
PREFIX='./hicdata/reordered_gb_hic_purged_masked'  ##prefix for index files
RAW_DIR='./raw_outdir'
FILT_DIR='./filt_outdir'
FILTER='./scripts/filter_five_end.pl'
COMBINER='./scripts/two_read_bam_combiner.pl'
STATS='./scripts/get_stats.pl'
PICARD='/opt/nesi/CS400_centos7_bdw/picard/2.26.10-Java-11.0.4/picard'
TMP_DIR='~/nobackup/source_files/hic/mapping/temp'
PAIR_DIR='./paired_outdir'
REP_DIR='./deduped_outdir'
REP_LABEL=${LABEL}_rep1
MAPQ_FILTER=10
CPU=12

$BWA mem -t $CPU $REF $IN_DIR/${SRA}1.fastq.gz | $SAMTOOLS view -@ $CPU -Sb - > $RAW_DIR/${SRA}_1.bam

echo "reads 1 complete"

$BWA mem -t $CPU $REF $IN_DIR/${SRA}2.fastq.gz | $SAMTOOLS view -@ $CPU -Sb - > $RAW_DIR/${SRA}_2.bam

echo "reads 2 complete"
```

Filter so only paired reads that map at 5' end retained:
```
#!/bin/bash -e
#SBATCH --job-name=fivefilter
#SBATCH --time=15:00:00
#SBATCH --mem=16G
#SBATCH --account=uoo02831
#SBATCH --cpus-per-task=12

module load BWA SAMtools 

SRA='TRIMMED_HiC_R'
LABEL='gb_hic_map'
IN_DIR='./hicdata'
BWA='/opt/nesi/CS400_centos7_bdw/BWA/0.7.18-GCC-12.3.0/bin/bwa'
REF='./hicdata/reordered_gb_hic_purged_masked'
FAIDX='$REF.fa.fai'
PREFIX='./hicdata/reordered_gb_hic_purged_masked'  ##prefix for index files
RAW_DIR='./raw_outdir'
FILT_DIR='./filt_outdir'
FILTER='./scripts/filter_five_end.pl'
COMBINER='./scripts/two_read_bam_combiner.pl'
STATS='./scripts/get_stats.pl'
PICARD='/opt/nesi/CS400_centos7_bdw/picard/2.26.10-Java-11.0.4/picard'
TMP_DIR='~/nobackup/source_files/hic/mapping/temp'
PAIR_DIR='./paired_outdir'
REP_DIR='./deduped_outdir'
REP_LABEL=${LABEL}_rep1
MAPQ_FILTER=10
CPU=12

echo "### Step 2.A: Filter 5' end (1st)"

samtools view -h $RAW_DIR/${SRA}_1.bam | perl $FILTER | samtools view -Sb > $FILT_DIR/${SRA}_1.bam
echo "read 1 complete"

echo "### Step 2.B: Filter 5' end (2nd)"

samtools view -h $RAW_DIR/${SRA}_2.bam | perl $FILTER | samtools view -Sb > $FILT_DIR/${SRA}_2.bam

echo "read 2 complete"
```

Pair reads:
```
#!/bin/bash -e
#SBATCH --job-name=fivefilter
#SBATCH --time=15:00:00
#SBATCH --mem=16G
#SBATCH --account=uoo02831
#SBATCH --cpus-per-task=12

module load BWA SAMtools 

SRA='TRIMMED_HiC_R'
LABEL='gb_hic_map'
IN_DIR='./hicdata'
BWA='/opt/nesi/CS400_centos7_bdw/BWA/0.7.18-GCC-12.3.0/bin/bwa'
REF='./hicdata/reordered_gb_hic_purged_masked'
FAIDX='$REF.fai'
PREFIX='./hicdata/reordered_gb_hic_purged_masked'  ##prefix for index files
RAW_DIR='./raw_outdir'
FILT_DIR='./filt_outdir'
FILTER='./scripts/filter_five_end.pl'
COMBINER='./scripts/two_read_bam_combiner.pl'
STATS='./scripts/get_stats.pl'
PICARD='/opt/nesi/CS400_centos7_bdw/picard/2.26.10-Java-11.0.4/picard'
TMP_DIR='~/nobackup/source_files/hic/mapping/temp'
PAIR_DIR='./paired_outdir'
REP_DIR='./deduped_outdir'
REP_LABEL=${LABEL}_rep1
MAPQ_FILTER=10
CPU=12


perl $COMBINER $FILT_DIR/${SRA}_1.bam $FILT_DIR/${SRA}_2.bam samtools $MAPQ_FILTER | samtools view -bS -t $FAIDX - | samtools sort -@ $CPU -o $TMP_DIR/$SRA.bam -

echo "combining complete"
```


Combine bams
```
#!/bin/bash -e
#SBATCH --job-name=combiner
#SBATCH --time=15:00:00
#SBATCH --mem=16G
#SBATCH --account=uoo02831
#SBATCH --cpus-per-task=12

module load BWA SAMtools picard

SRA='TRIMMED_HiC_R'
LABEL='gb_hic_map'
IN_DIR='./hicdata'
REF='./hicdata/reordered_gb_hic_purged_masked'
FAIDX='$REF.fa.fai'
PREFIX='./hicdata/reordered_gb_hic_purged_masked'  ##prefix for index files
RAW_DIR='./raw_outdir'
FILT_DIR='./filt_outdir'
FILTER='./scripts/filter_five_end.pl'
COMBINER='./scripts/two_read_bam_combiner.pl'
STATS='./scripts/get_stats.pl'
TMP_DIR='./temp'
PAIR_DIR='./paired_outdir'
REP_DIR='./deduped_outdir'
REP_LABEL=${LABEL}_rep1
MAPQ_FILTER=10
CPU=12

perl $COMBINER $FILT_DIR/${SRA}_1.bam $FILT_DIR/${SRA}_2.bam samtools $MAPQ_FILTER | samtools view -bS -t $FAIDX - | samtools sort -@ $CPU -o $TMP_DIR/$SRA.bam -

echo "combining complete"

```

Add read groups with picard
```
java -Xmx4G -Djava.io.tmpdir=temp/ -jar /opt/nesi/CS400_centos7_bdw/picard/2.26.10-Java-11.0.4/picard.jar AddOrReplaceReadGroups --INPUT $TMP_DIR/$SRA.bam --OUTPUT $PAIR_DIR/$SRA.bam --RGID $SRA --RGLB $SRA --RGPL ILLUMINA --RGSM $LABEL --RGPU none
```

Mark PCR duplicates:
```
java -Xmx30G -XX:-UseGCOverheadLimit -Djava.io.tmpdir=temp/ -jar $PICARD MarkDuplicates --INPUT $PAIR_DIR/$SRA.bam --OUTPUT $REP_DIR/$REP_LABEL.bam --METRICS_FILE $REP_DIR/metrics.$REP_LABEL.txt --TMP_DIR $TMP_DIR --ASSUME_SORTED TRUE --VALIDATION_STRINGENCY LENIENT --REMOVE_DUPLICATES TRUE
```
