## Jellyfish
uses k-mer ditribution to estimate genome size
```
jellyfish count -m 21 -s 100M -t 4 -C <(zcat gbrev_hifi_reads.fq.gz)
```
counts canonical(-C) 21-mers in the sequence file

plot histogram:
```
jellyfish histo -t 4 mer_counts.jf > gb_reads.histo
```
then input the first 500 lines of the histogram into [GenomeScope2](http://genomescope.org/genomescope2.0/analysis.php?code=S16GOND4CPJHKGhyQ1aM) <br>
**Genomscope outputs:**
```
GenomeScope version 2.0
input file = user_uploads/S16GOND4CPJHKGhyQ1aM
output directory = user_data/S16GOND4CPJHKGhyQ1aM
p = 2
k = 21

property                      min               max               
Homozygous (aa)               98.5482%          98.6751%          
Heterozygous (ab)             1.32492%          1.45181%          
Genome Haploid Length         315,572,173 bp    319,006,280 bp    
Genome Repeat Length          8,654,641 bp      8,748,823 bp      
Genome Unique Length          306,917,531 bp    310,257,458 bp    
Model Fit                     81.9721%          82.0721%          
Read Error Rate               0.344825%         0.344825%
```
haploid of 300mb - diploid of 600mb which is about the size of our assembly :) yay

![linear_plot](https://github.com/user-attachments/assets/b7acb9a7-8d30-4c0d-b0af-b8bb500dd9ce)

## BUSCO - estimate assembly completeness
```
 busco -i GB_full.p_ctg.fa -m geno -o gbrev --auto-lineage-euk
```
output:
```
# BUSCO version is: 5.6.1 
# The lineage dataset is: eukaryota_odb10 (Creation date: 2024-01-08, number of genomes: 70, number of BUSCOs: 255)
# Summarized benchmarking in BUSCO notation for file /scale_wlg_nobackup/filesets/nobackup/uoo02831/ashleigh/source_files/genome/Gbrev_assembly/asm1_purgedups/GB_full.p_ctg.fa
# BUSCO was run in mode: euk_genome_met
# Gene predictor used: metaeuk

	***** Results: *****

	C:99.6%[S:96.1%,D:3.5%],F:0.0%,M:0.4%,n:255	   
	254	Complete BUSCOs (C)			   
	245	Complete and single-copy BUSCOs (S)	   
	9	Complete and duplicated BUSCOs (D)	   
	0	Fragmented BUSCOs (F)			   
	1	Missing BUSCOs (M)			   
	255	Total BUSCO groups searched		   

Assembly Statistics: <- nothing under here?


Dependencies and versions:
	hmmsearch: 3.3
	bbtools: 39.01
	metaeuk: GITDIR-NOTFOUND  <- is this bad?
	busco: 5.6.1
```

## Running purge_dups (Lydia method/stepwise)

**1. minimap2 to align reads to reference** 
```
sbatch gbminimap2.sl
```
[gbminimap2.sl](gbminimap2.sl) - aligns and converts aligned sam to paf.gz for next steps
<br>
edit: i actually think it didnt convert to paf -a outputs as sam i think - so GB_aln.paf.gz is a sam i think?? just compressed
downstream steps all ran empty when using the output from this one so i reran it slightly differently
```
seff 50616280

output:
Job ID: 50616280
Cluster: mahuika
User/Group: iwias275/iwias275
State: COMPLETED (exit code 0)
Nodes: 1
Cores per node: 16
CPU Utilized: 4-04:58:41
CPU Efficiency: 56.79% of 7-09:48:00 core-walltime
Job Wall-clock time: 11:06:45
Memory Utilized: 10.92 GB
Memory Efficiency: 34.13% of 32.00 GB
```
if need to rerun can reduce memory/time

[minimap_paf.sl](minimap_paf.sl) - reran with xasm20 and with paf output?
<!--xasm20 for CSS reads - im honestly still unclear on if this was right or should stick with map-hifi but... whats done is done--> 
```
sbatch minimap_paf.sl

seff
Job ID: 50638268
Cluster: mahuika
User/Group: iwias275/iwias275
State: COMPLETED (exit code 0)
Nodes: 1
Cores per node: 16
CPU Utilized: 4-04:44:24
CPU Efficiency: 99.06% of 4-05:41:52 core-walltime
Job Wall-clock time: 06:21:22
Memory Utilized: 6.68 GB
Memory Efficiency: 20.89% of 32.00 GB
```
didnt reduce mem coz i got scared but def can
<!--not sure if this was strictly necessary - could have converted the sam manually? but idk how to do that so!-->

**2. stats for mapped long reads**
```
pbcstat GB_aln.paf.gz #produces PB.base.cov and PB.stat files

calcuts PB.stat > cutoffs 2>calcults.log
```
**3. split assembly and do self-self alignment?**
```
split_fa GB_full.p_ctg.fa > GB_asm.split

minimap2 -xasm5 -DP GB_asm.split GB_asm.split | gzip -c - > GB_asm.split.self.paf.gz
```
splits the assembly then aligns to itself??? whu

**4. purge haplotigs and duplicates**
```
purge_dups -2 -T cutoffs -c PB.base.cov GB_asm.split.self.paf.gz > dups.bed 2> purge_dups.log
```

**5. Get purged primary and haplotig sequences from draft assembly** - this step may have been unncecessary
```
get_seqs -e dups.bed GB_full.p_ctg.fa

OUTPUT:
[W::get_seqs_core] ptg000001l 3240744 3276904 is skipped
[W::get_seqs_core] ptg000039l 3478938 3546001 is skipped
[W::get_seqs_core] ptg000073l 1330939 1349834 is skipped
[W::get_seqs_core] ptg000099l 1182349 1251107 is skipped
[W::get_seqs_core] ptg000187l 202589 420058 is skipped
```
-e option means only removes duplications from ends of contigs - remove if want to take from middle as well but might delete false positive duplications

**output hap.fa**<br>
this file contains the purged sequences identified as haplotigs
```
gfastats hap.fa

OUTPUT:
+++Assembly summary+++: 
# scaffolds: 397
Total scaffold length: 28433761
Average scaffold length: 71621.56
Scaffold N50: 106848
Scaffold auN: 698905.64
Scaffold L50: 33
Largest scaffold: 3280424
Smallest scaffold: 12723
# contigs: 397
Total contig length: 28433761
Average contig length: 71621.56
Contig N50: 106848
Contig auN: 698905.64
Contig L50: 33
Largest contig: 3280424
Smallest contig: 12723
# gaps in scaffolds: 0
Total gap length in scaffolds: 0
Average gap length in scaffolds: 0.00
Gap N50 in scaffolds: 0
Gap auN in scaffolds: 0.00
Gap L50 in scaffolds: 0
Largest gap in scaffolds: 0
Smallest gap in scaffolds: 0
Base composition (A:C:G:T): 7128599:6979205:7035298:7290659
GC content %: 49.29
# soft-masked bases: 0
# segments: 397
Total segment length: 28433761
Average segment length: 71621.56
# gaps: 0
# paths: 397
```


**looking at purged.fa**<br>
this is the full assembly purged of haplotigs/duplicate sequence
```
gfastats purged.fa

OUTPUT:
+++Assembly summary+++: 
# scaffolds: 266
Total scaffold length: 603880988
Average scaffold length: 2270229.28
Scaffold N50: 8857421
Scaffold auN: 8856822.02
Scaffold L50: 24
Largest scaffold: 27992649
Smallest scaffold: 17764
# contigs: 266
Total contig length: 603880988
Average contig length: 2270229.28
Contig N50: 8857421
Contig auN: 8856822.02
Contig L50: 24
Largest contig: 27992649
Smallest contig: 17764
# gaps in scaffolds: 0
Total gap length in scaffolds: 0
Average gap length in scaffolds: 0.00
Gap N50 in scaffolds: 0
Gap auN in scaffolds: 0.00
Gap L50 in scaffolds: 0
Largest gap in scaffolds: 0
Smallest gap in scaffolds: 0
Base composition (A:C:G:T): 169868341:132259585:132197431:169555631
GC content %: 43.79
# soft-masked bases: 0
# segments: 266
Total segment length: 603880988
Average segment length: 2270229.28
# gaps: 0
# paths: 266
```
this is about the right size - not exactly sure which step this file came from though

## Rerunning BUSCO 
running BUSCO on the purged.fa file to see how it looks:
```
busco -i purged.fa -m geno -o gbrev_purged_BUSCO -l eukaryota_odb10
```
output is exactly the same as before purging

## purge_haplotigs
```
samtools view -bS GB_aln.sam | samtools sort -o GB_aln_sorted.bam -T tmp.ali
```
convert sam > sorted bam
```
purge_haplotigs hist -t 8 -b GB_aln_sorted.bam -g purged.fa
```
create read-depth histogram - analyse genome coverage
(this step failed - troubleshooting)
