Review coverage to see if purge_haplotigs isnt working right for some reason

```
samtools faidx purged.fa  # This generates a file called purged.fa.fai
```
creates index file for purged assembly

```
cut -f 1-2 purged.fa.fai > genome.txt
```
cut the first 2 cols of fa.fai to take the scaffold length and name > tab sep txt file

```
bedtools makewindows -g genome.txt -w 2000 > purged_genome_windows.bed
```
use makewindows to make windows frm txt file w lengths and scaffold names
This file will contain the windows of the genome. Each line will represent a window, specifying the chromosome (scaffold name), start position, and end position.

```
bedtools coverage -a purged_genome_windows.bed -b align_rerun.bam -mean > coverage_20k_windows.txt
```
output file containing the mean coverage per window. Each line gives the window's scaffold name, start and end positions, and the average coverage for that window

-b align_rerun is the bam file of aligned rreads back to the genome i think (might be using the wrong ver. filing is a mess i need to be shot)


running busco actinopterygii on hic 52167491 and purged 52184521 
