Review coverage to see if purge_haplotigs isnt working right for some reason

```
samtools faidx purged.fa  # This generates a file called purged.fa.fai
```
creates index file for purged assembly

```
cut -f 1-2 purged.fa.fai > genome.txt
```
cut the first 2 cols of fa.fai to take the scaffold length and name > tab sep txt file

```
bedtools makewindows -g genome.txt -w 20000 > purged_genome_windows.bed
```
use makewindows to make windows frm txt file w lengths and scaffold names
This file will contain the windows of the genome. Each line will represent a window, specifying the chromosome (scaffold name), start position, and end position.
 change to 20,000 then rerun below
```
bedtools coverage -a purged_genome_windows.bed -b align_rerun.bam -mean > coverage_20k_windows.txt
```
output file containing the mean coverage per window. Each line gives the window's scaffold name, start and end positions, and the average coverage for that window

-b align_rerun is the bam file of aligned rreads back to the genome i think (might be using the wrong ver. filing is a mess i need to be shot)


running busco actinopterygii on hic 52167491 and purged 52184521 

BLAST - run blast search on genome
```
blastn \
-task megablast \
-query purged.fa \
-db nt \
-outfmt '6 qseqid staxids bitscore std' \
-max_target_seqs 1 \
-max_hsps 1 \
-num_threads 8 \
-evalue 1e-25 \
-out purgedassembly_vs_nt_megablast.out
```
warning: examining 5 or more matches is recommended

running 52890658

blobltools
```
module load Miniconda3

##create conda environment
conda create -n blobtools
conda activate blobtools
conda install -c anaconda -c bioconda matplotlib docopt tqdm wget pyyaml git pysam

##when redoing do conda init before to let u go back in?
```
get nodesDB? - creating a local database that BlobTools can use to assign taxonomic information?
```
wget ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump.tar.gz -P data/
tar zxf data/taxdump.tar.gz -C data/ nodes.dmp names.dmp
./blobtools nodesdb --nodes data/nodes.dmp --names data/names.dmp
```
create to create blobDB?
```
./blobtools create -i ../purged.fa -b ../align_rerun.bam -t **example/blast.out** -o example/test && \
./blobtools view -i example/test.blobDB.json && \
./blobtools plot -i example/test.blobDB.json
```
need a hits file for -t and change the other bits from example/test when get

meryl - rerunning kmer analysis

```
#!/bin/bash -e
#SBATCH --account=uoo02831
#SBATCH --job-name=meryl
#SBATCH --time=02:00:00
#SBATCH --cpus-per-task=16
#SBATCH --mem=64G
#SBATCH --partition=milan

# Modules
module purge
module load Merqury/1.3-Miniconda3

# Run
meryl count k=30 memory=64 threads=16 \
  gbrev_hifi_reads.fq.gz \
  output gb-read-kmerdb.meryl
```
creates kmer db 

```
meryl histogram gb-read-kmerdb.meryl > gb-read-kmerdb.hist
```
creates kmer hist from db
place in genomescope2.0:

http://genomescope.org/genomescope2.0/analysis.php?code=ech9pLIBECiwO7EhVusf

have a look and see if it looks the same as the jellyfish one
the model fit still not rly great but idk whattttttt thagt meand for anything if im being honest

