## Minimap2

Rerun minimap2 on HiC primary assembly

```
#!/bin/bash -e
#SBATCH --job-name=gbminimaphic # Job name (shows up in the queue)
#SBATCH --time=20:00:00      # Walltime (HH:MM:SS) 48h
#SBATCH --mem=32G            # Memory in MB
#SBATCH --account=uoo02831
#SBATCH --cpus-per-task=16

module load minimap2/2.24-GCC-11.3.0
module load SAMtools/1.16.1-GCC-11.3.0

minimap2 \
    -t 16 -ax map-hifi \
    GB_full_HIC.p_ctg.fa gbrev_hifi_reads.fq.gz \
    --secondary=no | samtools sort -m 5G -o hic_asm_minimap_align.bam -T temp.ali \
    2> minimap2_hic_errors.log

echo "alignment and file conversion complete"

```
 need a paf file for purgedups
 ```
#!/bin/bash -e
#SBATCH --job-name=gbminimaphic # Job name (shows up in the queue)
#SBATCH --time=15:00:00      # Walltime (HH:MM:SS) 48h
#SBATCH --mem=32G            # Memory in MB
#SBATCH --account=uoo02831
#SBATCH --cpus-per-task=32

module load minimap2
module load pigz

minimap2 \
	-t 32 -x map-hifi \
	GB_full_HIC.p_ctg.fa gbrev_hifi_reads.fq.gz \
	> hic_minimap_align.paf \
	2> minimap_paf_errors.log

echo "alignment complete"

pigz -p 32 -c hic_minimap_align.paf > hic_minimap_align.paf.gz

echo "compression complete"

```


## Purge_dups

```
module load purge_dups
module load minimap2
```

```
pbcstat hic_minimap_align.paf

calcuts PB.stat > cutoffs 2>calcults.log
```
produces PB.base.cov and PB.stat files

split assembly and do self-self alignment
```
split_fa GB_full.p_ctg.fa > GB_asm.split

minimap2 -xasm5 -DP gb_hic_asm.split gb_hic_asm.split | gzip -c - > gb_hic_asm.split.self.paf.gz
```

identify haplotigs and duplicates - places coordinates of dup/hap regions into file dup_sequences.bed
```
purge_dups -2 -T cutoffs -c PB.base.cov gb_hic_asm.split.self.paf.gz > dup_sequences.bed 2> purge_dups.log
```
purge duplicates from full genome using the bed file above
```
get_seqs -e -p gb_hic_purged dup_sequences.bed GB_full_HIC.p_ctg.fa
```
-e = rm only dups at the end of contigs - remove for more stringent but might cut out actual stuff
produces purged.fa and hap.fa (with prefix) - gb_hic_purged.fa is final purged draft taken forwards
```
[W::get_seqs_core] ptg000001l 3240733 3273834 is skipped
[W::get_seqs_core] ptg000007l 302571 1899292 is skipped
[W::get_seqs_core] ptg000009l 3740010 3829306 is skipped
[W::get_seqs_core] ptg000037l 656985 796933 is skipped
[W::get_seqs_core] ptg000038l 3478936 3545999 is skipped
[W::get_seqs_core] ptg000100l 1293521 1362279 is skipped
[W::get_seqs_core] ptg000192l 202589 420872 is skipped
```
check gfastats of full and purged assemblies:
```
#GB_full_HIC.p_ctg.fa
+++Assembly summary+++: 
# scaffolds: 664
Total scaffold length: 630588521
Average scaffold length: 949681.51
Scaffold N50: 7215088
Scaffold auN: 8253254.37
Scaffold L50: 27
Largest scaffold: 28003021
Smallest scaffold: 12723
# contigs: 664
Total contig length: 630588521
Average contig length: 949681.51
Contig N50: 7215088
Contig auN: 8253254.37
Contig L50: 27
Largest contig: 28003021
Smallest contig: 12723
# gaps in scaffolds: 0
Total gap length in scaffolds: 0
Average gap length in scaffolds: 0.00
Gap N50 in scaffolds: 0
Gap auN in scaffolds: 0.00
Gap L50 in scaffolds: 0
Largest gap in scaffolds: 0
Smallest gap in scaffolds: 0
Base composition (A:C:G:T): 176460736:138981680:138809043:176337062
GC content %: 44.05
# soft-masked bases: 0
# segments: 664
Total segment length: 630588521
Average segment length: 949681.51
# gaps: 0
# paths: 664
```

```
#gb_hic_purged.fa
+++Assembly summary+++: 
# scaffolds: 278
Total scaffold length: 604930116
Average scaffold length: 2176007.61
Scaffold N50: 7780314
Scaffold auN: 8577137.58
Scaffold L50: 25
Largest scaffold: 28003021
Smallest scaffold: 17764
# contigs: 278
Total contig length: 604930116
Average contig length: 2176007.61
Contig N50: 7780314
Contig auN: 8577137.58
Contig L50: 25
Largest contig: 28003021
Smallest contig: 17764
# gaps in scaffolds: 0
Total gap length in scaffolds: 0
Average gap length in scaffolds: 0.00
Gap N50 in scaffolds: 0
Gap auN in scaffolds: 0.00
Gap L50 in scaffolds: 0
Largest gap in scaffolds: 0
Smallest gap in scaffolds: 0
Base composition (A:C:G:T): 170177384:132534868:132376761:169841103
GC content %: 43.79
# soft-masked bases: 0
# segments: 278
Total segment length: 604930116
Average segment length: 2176007.61
# gaps: 0
# paths: 278
```
way less contigs in the purged asm, also smaller - looks good

## Longstitch

```
longstitch run draft=gb_hic_purged reads=gbrev_hifi_reads t=8 out_prefix=gb_hic_fulldraft G=3.02465058e8 longmap=hifi
```
check this over but this is my guess of the parameters i need.  not sure if i should round the G= or not
This is taking ages - if doesnt run today run as  job tmrw

maybe also run the ark one? idk what that is but i do have a kmer database from meryl? idk
```
longstitch tigmint-ntLink-arks draft=gb_hic_purged reads=gbrev_hifi_reads kmer_db=(meryl kmer database) t=8 out_prefix=gb_hic_fulldraft_arks G=3.02465058e8 longmap=hifi
```

note: **dont** put the file extensions in for the draft/read files - adds them so looks for gb_hic_purged.fa.fa 
