## Snakemake pipeline (from Ludo)

first download the repo from github and upload to nesi
unzip
```
unzip RAD_Snakemake_single-main.zip
```

load modules 
```
module load snakemake FastQC BWA VCFtools Stacks SAMtools cutadapt
```

run test
```
cp example_data/barcodes.txt . # copies test barcodes to wd

snakemake --cores all filtered.recode.vcf # run the pipeline
```
this i think takes the snakefile and config file and runs the pipeline - outputs VCF file that we use

configure inputs for my samples
```
mode: "refmap" # "denovo" or "refmap"

raw_fastq: # single-end currently not supported
  reads: "data/SQ1146_CE3JBANXX_s_2_fastq.txt.gz"

cutadapt:
  adapter: "AGATCGGAAGAGC" # Sequence of the adapter
  length: "50" # Mininimum length for refmap, common length for denovo
  minimum_phred: "25"  # Changed '=' to ':'

genome: # only needed for refmap mode
  ref: "data/gb_hic_purged.fa"

vcf_filtering:
  parameters: "--max-missing 0.8 --maf 0.0001" # vcftools arguments, passed at once
```
