## Snakemake pipeline
[from ludo stuff](https://github.com/ldutoit/RAD_Snakemake_single.git)

first download the repo from github and upload to nesi
unzip
```
unzip RAD_Snakemake_single-main.zip
```

load modules 
```
module load snakemake FastQC BWA VCFtools Stacks SAMtools cutadapt
```

run test
```
cd RAD_Snakemake_single # dir with the configs and stuff for the example data
cp example_data/barcodes.txt . # copies test barcodes to wd

snakemake --dag filtered.recode.vcf | dot -Tsvg > dag.svg # create the graph of rules (??)
```
```
snakemake --cores all filtered.recode.vcf
```
this i think takes the snakefile and config file and runs the pipeline - outputs VCF file that we use

### now configure inputs for my samples and run for real
```
mode: "refmap" # "denovo" or "refmap"

raw_fastq: # single-end currently not supported
  reads: "SQ1146_CE3JBANXX_s_2_fastq.txt.gz"

cutadapt:
  adapter: "CCGAGATCGGAAGAGC" # Sequence of the adapter
  length: "50" # Mininimum length for refmap, common length for denovo
  minimum_phred: "25"  # Changed '=' to ':'

genome: # only needed for refmap mode
  ref: "gb_hic_purged.fa"

vcf_filtering:
  parameters: "--max-missing 0.8 --maf 0.0001" # vcftools arguments, passed at once
```

issue with BWA step:
```
[Wed Apr  9 13:09:35 2025]
rule bwa_map:
    input: gb_hic_purged.fa, samples/KOARO_756.5.fq.gz
    output: mapped_reads/KOARO_756.5.bam
    jobid: 367
    reason: Missing output files: mapped_reads/KOARO_756.5.bam; Input files updated by another job: samples/KOARO_756.5.fq.gz
    wildcards: sample=KOARO_756.5
    resources: tmpdir=/dev/shm/jobs/54015074

[E::bwa_idx_load_from_disk] fail to locate the index files
[E::bwa_idx_load_from_disk] fail to locate the index files
[E::bwa_idx_load_from_disk] fail to locate the index files
[E::bwa_idx_load_from_disk] fail to locate the index files
[main_samview] fail to read the header from "-".
[main_samview] fail to read the header from "-".
[main_samview] fail to read the header from "-".
[main_samview] fail to read the header from "-".
[Wed Apr  9 13:09:35 2025]
[Wed Apr  9 13:09:35 2025]
[Wed Apr  9 13:09:35 2025]
Error in rule bwa_map:
    jobid: 319
    input: gb_hic_purged.fa, samples/KOARO_1663.3.fq.gz
    output: mapped_reads/KOARO_1663.3.bam
    shell:
        bwa mem gb_hic_purged.fa samples/KOARO_1663.3.fq.gz | samtools view -Sb - > mapped_reads/KOARO_1663.3.bam
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

[Wed Apr  9 13:09:35 2025]
Error in rule bwa_map:
    jobid: 309
    input: gb_hic_purged.fa, samples/KOARO_831.3.fq.gz
    output: mapped_reads/KOARO_831.3.bam
    shell:
        bwa mem gb_hic_purged.fa samples/KOARO_831.3.fq.gz | samtools view -Sb - > mapped_reads/KOARO_831.3.bam
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Error in rule bwa_map:
    jobid: 367
    input: gb_hic_purged.fa, samples/KOARO_756.5.fq.gz
    output: mapped_reads/KOARO_756.5.bam
    shell:
        bwa mem gb_hic_purged.fa samples/KOARO_756.5.fq.gz | samtools view -Sb - > mapped_reads/KOARO_756.5.bam
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Error in rule bwa_map:
    jobid: 313
    input: gb_hic_purged.fa, samples/KOARO_831.5.fq.gz
    output: mapped_reads/KOARO_831.5.bam
    shell:
        bwa mem gb_hic_purged.fa samples/KOARO_831.5.fq.gz | samtools view -Sb - > mapped_reads/KOARO_831.5.bam
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Removing output files of failed job bwa_map since they might be corrupted:
mapped_reads/KOARO_1663.3.bam
Removing output files of failed job bwa_map since they might be corrupted:
mapped_reads/KOARO_831.3.bam
Removing output files of failed job bwa_map since they might be corrupted:
mapped_reads/KOARO_756.5.bam
Removing output files of failed job bwa_map since they might be corrupted:
mapped_reads/KOARO_831.5.bam
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2025-04-09T110704.964358.snakemake.log
```

issue could be with indexing?  says it cant read the index file.........

run index command separately:
```
bwa index -p gb_hic_purged.fa gb_hic_purged.fa
```
```
self note - outputs:
gb_hic_purged.fa.amb
gb_hic_purged.fa.sa
gb_hic_purged.fa.pac
gb_hic_purged.fa.bwt
gb_hic_purged.fa.ann
```
this worked now it moves on to mem step but i dont understand why it didnt work to begin with

finished!
```
[Thu Apr 10 15:51:54 2025]
Finished job 0.
385 of 385 steps (100%) done
Complete log: .snakemake/log/2025-04-10T114012.353765.snakemake.log
```
## filtering

look at the missing data
```
sort -k 4n filtered.imiss
```
sorts based on row 4 which is the number of missing snps?

```
INDV    N_DATA  N_GENOTYPES_FILTERED    N_MISS  F_MISS
KOARO_1663.3    116756  0       422     0.00361438
KOARO_163.7     116756  0       465     0.00398266
KOARO_163.11    116756  0       473     0.00405118
KOARO_CHL59     116756  0       539     0.00461647
KOARO_CHL65     116756  0       560     0.00479633
KOARO_163.6     116756  0       578     0.0049505
KOARO_CHL53     116756  0       592     0.0050704
KOARO_CHL58     116756  0       592     0.0050704
KOARO_CHL52     116756  0       625     0.00535304
KOARO_16310     116756  0       637     0.00545582
KOARO_13Mile13  116756  0       678     0.00580698
KOARO_13Mile19  116756  0       696     0.00596115
KOARO_CHL54     116756  0       698     0.00597828
KOARO_13Mile20  116756  0       705     0.00603823
KOARO_163.8     116756  0       705     0.00603823
KOARO_CHL61     116756  0       708     0.00606393
KOARO_13Mile12  116756  0       720     0.00616671
KOARO_12Mile7   116756  0       728     0.00623523
KOARO_GB323     116756  0       767     0.00656926
KOARO_CHL02     116756  0       770     0.00659495
KOARO_GB330     116756  0       772     0.00661208
KOARO_GB319     116756  0       795     0.00680907
KOARO_13Mile18  116756  0       818     0.00700606
KOARO_GB326     116756  0       824     0.00705745
KOARO_12Mile10  116756  0       831     0.00711741
KOARO_CHL57     116756  0       841     0.00720306
KOARO_MarK07    116756  0       856     0.00733153
KOARO_163.9     116756  0       864     0.00740005
KOARO_CHL56     116756  0       886     0.00758848
KOARO_12Mile12  116756  0       887     0.00759704
KOARO_MarK11    116756  0       891     0.0076313
KOARO_Bre_26    116756  0       904     0.00774264
KOARO_GB331     116756  0       909     0.00778547
KOARO_LSL64     116756  0       912     0.00781116
KOARO_Par_7     116756  0       913     0.00781973
KOARO_MarK04    116756  0       918     0.00786255
KOARO_LSL59     116756  0       947     0.00811093
KOARO_GRN_01    116756  0       957     0.00819658
KOARO_SYL25     116756  0       985     0.0084364
KOARO_MarK12    116756  0       991     0.00848779
KOARO_CHL66     116756  0       996     0.00853061
KOARO_GRN_K4    116756  0       997     0.00853918
KOARO_GB327     116756  0       998     0.00854774
KOARO_Par_16    116756  0       1017    0.00871047
KOARO_GRN_51    116756  0       1021    0.00874473
KOARO_MarK10    116756  0       1031    0.00883038
KOARO_GRN_55    116756  0       1038    0.00889034
KOARO_Mar-04    116756  0       1044    0.00894172
KOARO_LSL70     116756  0       1058    0.00906163
KOARO_GRN_57    116756  0       1064    0.00911302
KOARO_MarK02    116756  0       1079    0.0092415
KOARO_SYL52     116756  0       1081    0.00925862
KOARO_MarK05    116756  0       1085    0.00929288
KOARO_Mar-06    116756  0       1111    0.00951557
KOARO_LSL60     116756  0       1119    0.00958409
KOARO_756.5     116756  0       1135    0.00972113
KOARO_GRN_K2    116756  0       1144    0.00979821
KOARO_Par_10    116756  0       1157    0.00990955
KOARO_Par_6     116756  0       1168    0.0100038
KOARO_13Mile31  116756  0       1182    0.0101237
KOARO_GRN_K_11  116756  0       1188    0.0101751
KOARO_MarK14    116756  0       1195    0.010235
KOARO_MarK08    116756  0       1198    0.0102607
KOARO_Mar-05    116756  0       1210    0.0103635
KOARO_12Mile14  116756  0       1222    0.0104663
KOARO_Lag_K15   116756  0       1245    0.0106633
KOARO_Lag_K19   116756  0       1246    0.0106718
KOARO_SYL14     116756  0       1267    0.0108517
KOARO_Lag_K14   116756  0       1290    0.0110487
KOARO_163.1     116756  0       1292    0.0110658
KOARO_Par_5     116756  0       1308    0.0112029
KOARO_GRN_53    116756  0       1309    0.0112114
KOARO_Lag_K01   116756  0       1349    0.011554
KOARO_SYL07     116756  0       1358    0.0116311
KOARO_Par_13    116756  0       1359    0.0116397
KOARO_Lag_K09   116756  0       1361    0.0116568
KOARO_Lag_K05   116756  0       1375    0.0117767
KOARO_163.2     116756  0       1380    0.0118195
KOARO_Lag_K08   116756  0       1391    0.0119137
KOARO_Lag_K13   116756  0       1392    0.0119223
KOARO_12Mile6   116756  0       1407    0.0120508
KOARO_Par_15    116756  0       1415    0.0121193
KOARO_Lag_K20   116756  0       1424    0.0121964
KOARO_Lag_K11   116756  0       1430    0.0122478
KOARO_Par_14    116756  0       1437    0.0123077
KOARO_Lag_K04   116756  0       1443    0.0123591
KOARO_Lag_K07   116756  0       1473    0.0126161
KOARO_GRN_K5    116756  0       1476    0.0126417
KOARO_Par_8     116756  0       1494    0.0127959
KOARO_12Mile25  116756  0       1579    0.0135239
KOARO_756.2     116756  0       1613    0.0138151
KOARO_GRN_K6    116756  0       1616    0.0138408
KOARO_756.6     116756  0       1690    0.0144746
KOARO_MarK01    116756  0       1731    0.0148258
KOARO_756.8     116756  0       1755    0.0150313
KOARO_GRN_K7    116756  0       1782    0.0152626
KOARO_12Mile20  116756  0       1786    0.0152969
KOARO_GB320     116756  0       1792    0.0153482
KOARO_GB318     116756  0       1853    0.0158707
KOARO_Par_9     116756  0       1864    0.0159649
KOARO_Bre_17    116756  0       2020    0.017301
KOARO_12Mile27  116756  0       2105    0.0180291
KOARO_756.1     116756  0       2120    0.0181575
KOARO_12Mile15  116756  0       2243    0.019211
KOARO_GB324     116756  0       2264    0.0193909
KOARO_13Mile28  116756  0       2298    0.0196821
KOARO_Par_4     116756  0       2465    0.0211124
KOARO_Par_12    116756  0       2481    0.0212494
KOARO_831.4     116756  0       2497    0.0213865
KOARO_Lag_k02   116756  0       2549    0.0218319
KOARO_830.1     116756  0       2601    0.0222772
KOARO_830.3     116756  0       2629    0.022517
KOARO_830.2     116756  0       2640    0.0226113
KOARO_831.1     116756  0       2670    0.0228682
KOARO_831.5     116756  0       2682    0.022971
KOARO_831.3     116756  0       2693    0.0230652
KOARO_830.5     116756  0       2728    0.023365
KOARO_831.2     116756  0       2765    0.0236819
KOARO_13Mile30  116756  0       2854    0.0244441
KOARO_830.4     116756  0       2946    0.0252321
KOARO_GB317     116756  0       2970    0.0254377
KOARO_GB321     116756  0       3059    0.0261999
KOARO_Mar-12    116756  0       3324    0.0284696
KOARO_13Mile24  116756  0       3390    0.0290349
KOARO_GRN_K9    116756  0       3448    0.0295317
KOARO_13Mile5   116756  0       3524    0.0301826
KOARO_LSL67     116756  0       3766    0.0322553
KOARO_12Mile5   116756  0       3873    0.0331717
KOARO_Par_11    116756  0       3941    0.0337542
KOARO_GB329     116756  0       3972    0.0340197
KOARO_Bre_29    116756  0       3986    0.0341396
KOARO_SYL17     116756  0       4365    0.0373857
KOARO_756.9     116756  0       4368    0.0374114
KOARO_GRN_K3    116756  0       4655    0.0398695
KOARO_12Mile24  116756  0       4858    0.0416081
KOARO_Lag_K03   116756  0       5185    0.0444089
KOARO_GRN_K8    116756  0       5366    0.0459591
KOARO_756.3     116756  0       5426    0.046473
KOARO_SYL21     116756  0       5930    0.0507897
KOARO_GRN_K10   116756  0       5939    0.0508668
KOARO_Par_18    116756  0       5952    0.0509781
KOARO_GRN_56    116756  0       6368    0.0545411
KOARO_12Mile21  116756  0       6911    0.0591918
KOARO_756.4     116756  0       7124    0.0610161
KOARO_756.7     116756  0       7665    0.0656497
KOARO_13Mile23  116756  0       8106    0.0694268
KOARO_75610     116756  0       8163    0.069915
KOARO_Bre_15    116756  0       8416    0.0720819
KOARO_Bre_13    116756  0       9741    0.0834304
KOARO_13Mile9   116756  0       10355   0.0886892
KOARO_Par_2     116756  0       10517   0.0900767
KOARO_12Mile17  116756  0       10847   0.0929031
KOARO_Par_17    116756  0       10966   0.0939224
KOARO_Bre_5     116756  0       11999   0.10277
KOARO_Bre_3     116756  0       13191   0.112979
KOARO_13Mile11  116756  0       13303   0.113938
KOARO_Mar-15    116756  0       13702   0.117356
KOARO_Par_1     116756  0       15654   0.134074
KOARO_LSL15     116756  0       17562   0.150416
KOARO_Mar-08    116756  0       17819   0.152617
KOARO_13Mile8   116756  0       18965   0.162433
KOARO_Bre_2     116756  0       21335   0.182732
KOARO_12Mile4   116756  0       26057   0.223175
KOARO_12Mile26  116756  0       29060   0.248895
KOARO_Bre_8     116756  0       30279   0.259336
KOARO_Bre_9     116756  0       32311   0.27674
KOARO_CHL16     116756  0       32336   0.276954
KOARO_Bre_12    116756  0       34003   0.291231
KOARO_Mar-16    116756  0       36230   0.310305
KOARO_LSL76     116756  0       36353   0.311359
KOARO_Bre_10    116756  0       37052   0.317346
KOARO_12Mile1   116756  0       41730   0.357412
KOARO_LSL05     116756  0       41843   0.35838
KOARO_CHL72     116756  0       47331   0.405384
KOARO_LSL75     116756  0       59805   0.512222
KOARO_Mar-14    116756  0       62490   0.535219
KOARO_Bre_11    116756  0       65165   0.55813
KOARO_LSL74     116756  0       67486   0.578009
KOARO_CHL69     116756  0       67594   0.578934
KOARO_Par_3     116756  0       68865   0.58982
KOARO_CHL68     116756  0       71800   0.614958
KOARO_12Mile19  116756  0       78465   0.672043
KOARO_CHL13     116756  0       80851   0.692478
KOARO_Bre_7     116756  0       80991   0.693677
KOARO_LSL53     116756  0       81246   0.695861
KOARO_LSL52     116756  0       89141   0.763481
KOARO_CHL18     116756  0       93620   0.801843
GBSNEG1 116756  0       103501  0.886473  ## negative ctrl (but has 10% of snps still - which is bad)
GBSNEG2 116756  0       115465  0.988943  ## negative ctrl
GBSNEG4 116756  0       115481  0.98908   ## negative ctrl
KOARO_CHL10     116756  0       116263  0.995778
```
so now want to remove the snps present in the negative ctrl from all the other samples and also remove koaro_chl10

extract the negative controls into their own vcf file
```
vcftools --vcf filtered.recode.vcf --out negative.controls --indv GBSNEG1 --indv GBSNEG2 --indv GBSNEG4 --recode
```
remove snps that are not present in any of the tthree samples
```
vcftools --vcf negative.controls.recode.vcf --out negative.controls.nonmissing --max-missing 0.01 --recode
```
-max-missing 0.01 - maxmissing 0 allows all missing data, 1 allows none, so if i put 0.01 hopefully it removes the ones that have none for any but none of the ones that arent missing for at least 1 sample?

remove header
```
grep -v "##" negative.controls.nonmissing.recode.vcf > negative.ctrl.snps.nohead.txt
```

cut lines 1 and 2 - chrom and position
```
cut -f 1,2 negative.ctrl.snps.nohead.txt > snps.to.rm.txt
```
can also pipe this so its 1 cmd:
```
grep -v "##" negative.controls.nonmissing.recode.vcf | cut -f 1,2 > snps.to.rm.pipetest.txt
```

create list of indivs to rm:
rmindiv.txt
```
GBSNEG1
GBSNEG2
GBSNEG4
KOARO_CHL10
```

use these files to filter snps present in neg ctrls and remove negatives + 1 sample with really bad data
```
vcftools --vcf filtered.recode.vcf --out snpfiltered --exclude-positions snps.to.rm.txt --remove rmindiv.txt --max-missing 0.8 --recode 
```
