## Snakemake pipeline
[from ludo stuff](https://github.com/ldutoit/RAD_Snakemake_single.git)

first download the repo from github and upload to nesi
unzip
```
unzip RAD_Snakemake_single-main.zip
```

load modules 
```
module load snakemake FastQC BWA VCFtools Stacks SAMtools cutadapt
```

run test
```
cd RAD_Snakemake_single # dir with the configs and stuff for the example data
cp example_data/barcodes.txt . # copies test barcodes to wd

snakemake --dag filtered.recode.vcf | dot -Tsvg > dag.svg # create the graph of rules (??)
```
```
snakemake --cores all filtered.recode.vcf
```
this i think takes the snakefile and config file and runs the pipeline - outputs VCF file that we use

now configure inputs for my samples
```
mode: "refmap" # "denovo" or "refmap"

raw_fastq: # single-end currently not supported
  reads: "SQ1146_CE3JBANXX_s_2_fastq.txt.gz"

cutadapt:
  adapter: "AGATCGGAAGAGC" # Sequence of the adapter
  length: "50" # Mininimum length for refmap, common length for denovo
  minimum_phred: "25"  # Changed '=' to ':'

genome: # only needed for refmap mode
  ref: "gb_hic_purged.fa"

vcf_filtering:
  parameters: "--max-missing 0.8 --maf 0.0001" # vcftools arguments, passed at once
```

issue with BWA step:
```
[Wed Apr  9 13:09:35 2025]
rule bwa_map:
    input: gb_hic_purged.fa, samples/KOARO_756.5.fq.gz
    output: mapped_reads/KOARO_756.5.bam
    jobid: 367
    reason: Missing output files: mapped_reads/KOARO_756.5.bam; Input files updated by another job: samples/KOARO_756.5.fq.gz
    wildcards: sample=KOARO_756.5
    resources: tmpdir=/dev/shm/jobs/54015074

[E::bwa_idx_load_from_disk] fail to locate the index files
[E::bwa_idx_load_from_disk] fail to locate the index files
[E::bwa_idx_load_from_disk] fail to locate the index files
[E::bwa_idx_load_from_disk] fail to locate the index files
[main_samview] fail to read the header from "-".
[main_samview] fail to read the header from "-".
[main_samview] fail to read the header from "-".
[main_samview] fail to read the header from "-".
[Wed Apr  9 13:09:35 2025]
[Wed Apr  9 13:09:35 2025]
[Wed Apr  9 13:09:35 2025]
Error in rule bwa_map:
    jobid: 319
    input: gb_hic_purged.fa, samples/KOARO_1663.3.fq.gz
    output: mapped_reads/KOARO_1663.3.bam
    shell:
        bwa mem gb_hic_purged.fa samples/KOARO_1663.3.fq.gz | samtools view -Sb - > mapped_reads/KOARO_1663.3.bam
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

[Wed Apr  9 13:09:35 2025]
Error in rule bwa_map:
    jobid: 309
    input: gb_hic_purged.fa, samples/KOARO_831.3.fq.gz
    output: mapped_reads/KOARO_831.3.bam
    shell:
        bwa mem gb_hic_purged.fa samples/KOARO_831.3.fq.gz | samtools view -Sb - > mapped_reads/KOARO_831.3.bam
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Error in rule bwa_map:
    jobid: 367
    input: gb_hic_purged.fa, samples/KOARO_756.5.fq.gz
    output: mapped_reads/KOARO_756.5.bam
    shell:
        bwa mem gb_hic_purged.fa samples/KOARO_756.5.fq.gz | samtools view -Sb - > mapped_reads/KOARO_756.5.bam
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Error in rule bwa_map:
    jobid: 313
    input: gb_hic_purged.fa, samples/KOARO_831.5.fq.gz
    output: mapped_reads/KOARO_831.5.bam
    shell:
        bwa mem gb_hic_purged.fa samples/KOARO_831.5.fq.gz | samtools view -Sb - > mapped_reads/KOARO_831.5.bam
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Removing output files of failed job bwa_map since they might be corrupted:
mapped_reads/KOARO_1663.3.bam
Removing output files of failed job bwa_map since they might be corrupted:
mapped_reads/KOARO_831.3.bam
Removing output files of failed job bwa_map since they might be corrupted:
mapped_reads/KOARO_756.5.bam
Removing output files of failed job bwa_map since they might be corrupted:
mapped_reads/KOARO_831.5.bam
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2025-04-09T110704.964358.snakemake.log
```

issue could be with indexing?  says it cant read the index file.........

run index command separately:
```
bwa index -p gb_hic_purged.fa gb_hic_purged.fa
```
```
self note - outputs:
gb_hic_purged.fa.amb
gb_hic_purged.fa.sa
gb_hic_purged.fa.pac
gb_hic_purged.fa.bwt
gb_hic_purged.fa.ann
```
this worked now it moves on to mem step but i dont understand why it didnt work to begin with
