## Using BayeScan to do a fenome scan using FST for loci that differ btwn mig and nonmig pops

First want to refilter VCF to remove the CHatham/Auck Islands pops and the msterious GB samples:
```
add code here - nesi gone down
just using vcftools again but with a new indivs to rm file
```
Put in TASSEL (why) to have a look and filter MAF 
```
filter > filter genotype table sites > site min allele frequency = 0.01

resave VCF with this filtering

can also do this directly in vcftools when refiltering but did it this way because we tried to run PGDSpider with the full dataset and it said it was too big and this is how graham does it
```
Then want to convert this new VCF to a Bayes file using PGDSpider3
```
GUI but this is what I did:

Input file: snpfiltered2name.recode.vcf
Output format: GESTE/Bayescan
Output file: bayescan_result (i think... use a more informative name nxt time)

Press convert - comes up with a little screen where you can add a metadata file with pop IDs and pop assignments
- no headers, txt file
```

Take converted output and fed it into BayeScan GUI ver.
```
Input: bayescan_result

Output: bayescan_output / bayescan_small_output

Ran one with default parameters, one with reduced sample size, pilot runs, pilot run length
(reduced all from 5000 to 500)
```

## load into R - make some plots and stuff

**ADD BAYESCAN FST PLOTTING RMD HERE**
for this I needed a metadata file with the CHROM, POS, AND SNP ID columns from my VCF file so I could figure out which SNP is which in my FSoutput which has all of the snps in the same order so I did:
```
 grep -v "#"(name).recode.vcf | cut -f 1,2,3 > positionssnpbayesfiltered.txt
```
and then downloaded that file to bind in r

then i did my r stuff and output a text file that had all the significant chromosome positions and used that back in TASSEL to filter the VCF and view it
```
filter > filter genotype table sites > Chr Pos File - add the text file here
```

then have a new VCF with only the significant positions i can save and also can view graphically
![image](https://github.com/user-attachments/assets/1df8ccb7-1dbe-4a1f-9d7b-a8a2437128f0)

this file I called **BAYESCAN_SIGNIFICANTPOS.vcf**

now i am waiting to do geneious i think? idk

next i want to do some blasting of the regions with the signif snps and also some annotation to find out whats going on in the regions and also i think i should be aligning the SNPs to the genome somehow in some viewer or smth so I can look at where exactly they are to refine the
blasting sequences but idk how to do that also no one else seems to think that som maybe its not important

but def ask ludo what he thinks tomorrow

ok also did this with the full bayescan run but it looks like it was exactly the same - which is good :)  saved in fullbayescan_processing.rmd

## proper version (with masked data)

did same as above pretty much but filtered in bash rather than tassel

```
grep -v "#" maskedrefiltered_maf.recode.vcf | cut -f 1,2,3 > maskedgenome_bayescan_positions.txt
```
