Refilter to include the islands but not the mystery pops 

```
vcftools --vcf ../filtered.recode.vcf --exclude-positions ../snps.to.rm.txt --remove rmindiv_allsamp.txt --max-missing 0.8 --out islands_incl_full --recode
```
this is from the masked rerun snp calling 

rmindv_allsamp includes:
```
## negative ctrls
GBSNEG1
GBSNEG2
GBSNEG4
## mystery samples
KOARO_GB317
KOARO_GB318
KOARO_GB319
KOARO_GB320
KOARO_GB321
KOARO_GB323
KOARO_GB324
KOARO_GB326
KOARO_GB327
KOARO_GB329
KOARO_GB330
KOARO_GB331
## low data indvs
KOARO_CHL10
KOARO_LSL75
KOARO_Mar-14
KOARO_Bre_11
KOARO_LSL74
KOARO_CHL69
KOARO_Par_3
KOARO_CHL68
KOARO_12Mile19
KOARO_CHL13
KOARO_LSL53
KOARO_Bre_7
KOARO_LSL52
KOARO_CHL18

## gb samples, -ve ctrls, lowdata indvs
```

actually decied to keep the mystery samples and assume they are rotoroa so heres

rmindiv_only_lowdata.txt:
```
## negative ctrls
GBSNEG1
GBSNEG2
GBSNEG4
## low data indvs
KOARO_CHL10
KOARO_LSL75
KOARO_Mar-14
KOARO_Bre_11
KOARO_LSL74
KOARO_CHL69
KOARO_Par_3
KOARO_CHL68
KOARO_12Mile19
KOARO_CHL13
KOARO_LSL53
KOARO_Bre_7
KOARO_LSL52
KOARO_CHL18
```
refilter for allsamples.
```
Parameters as interpreted:
	--vcf ../filtered.recode.vcf
	--exclude-positions ../snps.to.rm.txt
	--remove rmindiv_only_lowdata.txt
	--max-missing 0.8
	--out all_samples
	--recode
```
now were gonna take this to iqtree first (use all the snps to construct the tree)  then filter it for maf 0.01 and take it thru pgdspider/bayescan

first cut the positions of each snp from the vcf and put them in a bed file
```
awk 'BEGIN {OFS="\t"} !/^#/ {print $1, $2-1, $2}' all_samples.recode.vcf > allsample_vcf_positions.bed
```
need to convert from 1base to 0base so do $2-1 - remove 1 from the position

now to frankenstein together some code from gemma/internet:

```
	## extract BED positions from masked genome fasta
bedtools getfasta -fi gb_hic_purged.fa.masked -bed allsample_vcf_positions.bed -fo reference_extracted_sequences.fasta

	## the resulting file has a header and each base on a different line
	##remove headers - resulting file still has a base on each line
awk '/^>/ {if (seq) print seq; seq=""; next} {seq=seq $0} END {print seq}' reference_extracted_sequences.fasta > refseq_string.txt

	##stick all the lines together - creates a little loop in sed to loop back to 'a' until the last line of the file ($!)
	##and tells it to put the next line up with the first and remove all newline characters then sticks Tasmanian_reference_GB in front 
sed ':a;N;$!ba;s/\n//g' refseq_string.txt | sed "s/^/Tasmanian_reference_GB\t/g" > refseq_string_row.txt
```
now we have a file with a name and then the reference sequence genotypes on it that we can cat on to the end of our phylip file
```
python vcf2phylip.py --input all_samples.recode.vcf

cat refseq_string_row.txt >> all_samples.recode.min4.phy
```
then manually change sample ## at top from 174 to 175
